scripts/release.sh

```diff
@@ -314,9 +330,41 @@ main() {
     exit 1
   fi
 
-  # TODO: signing process
+  local gh_repo
+  if [ "${REPOSITORY}" = "$(pwd)" ]; then
+    gh_repo=$(git remote get-url origin)
+  else
+    gh_repo="${REPOSITORY}"
+  fi
+
+  gh_repo=$(echo "${gh_repo}" | sed 's/^[^@]\+@//' | sed 's/https\?:\/\///' | sed 's/\.git$//' | tr ':' '/')
+  log_callout "Creating GitHub release for ${RELEASE_VERSION} on ${gh_repo}"
+
+  # Disable sellcheck SC2016, the single quoted syntax for sed is intentional.
+  # shellcheck disable=SC2016
+  local release_notes_temp_file=$(mktemp)
+  sed 's/${RELEASE_VERSION}/'"${RELEASE_VERSION}"'/' ./scripts/release_notes.tpl.txt > "${release_notes_temp_file}"
+  maybe_run gh release create "${RELEASE_VERSION}" \
+      --repo "${gh_repo}" \
+      --draft \
+      --prerelease \
```

Comment:
Do we want to be marking as pre-release?  I think this would depend on which branch we were creating a release for?  I.e. if we were publishing an alpha/rc for 3.6.0 vs publishing a patch release for 3.5 or 3.4.

Reply:
For `main`, it should be `--prerelease`, for `release-3.5`, it should be `--latest` and for `release-3.4`, it should not have an additional option.

I'm open to documenting it or adding a branch check. Let me know your thoughts.

Reply:
I think we need to have some conditional logic in the script to provide the appropriate parameter based on which branch we are performing release steps for.  Otherwise we would need to edit the instructions to make sure we update draft release manually?

Reply:
Sounds like a good idea, I'll implement it.